OpenRA.Mods.Common/Traits/AutoCarryable.cs

```diff
@@ -0,0 +1,159 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Can be carried by units with the trait `Carryall`.")]
+	public class AutoCarryableInfo : CarryableInfo
+	{
+		[Desc("Required distance away from destination before requesting a pickup. Default is 6 cells.")]
+		public readonly WDist MinDistance = WDist.FromCells(6);
+
+		public override object Create(ActorInitializer init) { return new AutoCarryable(init.Self, this); }
+	}
+
+	public class AutoCarryable : Carryable, INotifyHarvesterAction, ICallForTransport
```

Comment:
This can be renamed to something a bit more descriptive like `RequestsAutoCarryall`.